derivative.go

```diff
@@ -125,7 +126,7 @@ func (n *DerivativeNode) derivative(prev, curr models.Fields, prevTime, currTime
 	f1, ok := numToFloat(curr[n.d.Field])
 	if !ok {
 		n.incrementErrorCount()
-		n.logger.Printf("E! cannot apply derivative to type %T", curr[n.d.Field])
+		n.diag.Error("cannot perform derivative", fmt.Errorf("field %s is the wrong %T", n.d.Field, curr[n.d.Field]))
```

Comment:
I think this one should read:

```
n.diag.Error("cannot perform derivative", errors.New("field is the wrong type") keyvalue.KV("field",n.d.Field), keyvalue.KV("type", reflect.TypeOf(curr[n.d.Field]).String()))
```

Reply:
```go
reflect.TypeOf(curr[n.d.Field]).String()
``` 
panics when `curr[n.d.Field]` is nil
https://play.golang.org/p/gjlr3kuusL

Went with
```go
keyvalue.KV("type", fmt.Sprintf("%T", curr[n.d.Field]))
```
instead